code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+	var/hg_mult = 1										//multiplier for health gained per tick when on dark tile
+	var/hd_mult = 3									 //multiplier for health drained per tick on bright tile
+//	var/show_desc = TRUE										   //For the ability menu
+
+	var/lifestage=3												 //1=baby grue, 2=grueling, 3=(mature) grue
+	var/eatencount=0												//number of sentient carbons eaten, makes the grue more powerful
+	var/eatencharge=0												//power charged by eating sentient carbons, increments with eatencount but is spent on upgrades
+	var/dark_dim_light=0 //darkness level currently the grue is currently exposed to, 0=nice and dark, 1=passably dim, 2=too bright
+	var/busy=0 //busy laying an egg
+
+
+	//eyesight related stuff
+	see_in_dark = 8
+	//keeping this here for later color matrix testing
+	var/a_use_alpha=1
+	var/a_blend_add_test=255
+
+	var/a_matrix_testing_override = 0 //not used for now
+	var/a_11 = 1
+	var/a_12 = 0
+	var/a_13 = 0
+	var/a_14 = 1
+	var/a_21 = -1
+	var/a_22 = 0.2
+	var/a_23 = 0.2
+	var/a_24 = 1
+	var/a_31 = -1
+	var/a_32 = 0.2
+	var/a_33 = 0.2
+	var/a_34 = 1
+	var/a_41 = 0
+	var/a_42 = 0
+	var/a_43 = 0
+	var/a_44 = 1
+	var/a_51 = 0
+	var/a_52 = 0
+	var/a_53 = 0
+	var/a_54 = 0
+
+//	var/b_matrix_testing_override=0 //not used for now
+//	var/b_11 = 1
+//	var/b_12 = 0
+//	var/b_13 = 0
+//	var/b_21 = 0
+//	var/b_22 = 1
+//	var/b_23 = 0
+//	var/b_31 = 0
+//	var/b_32 = 0
+//	var/b_33 = 1
+//	var/b_41 = 0
+//	var/b_42 = 0
+//	var/b_43 = 0
+
+/mob/living/simple_animal/hostile/grue/regular_hud_updates()
+	..()
+	if(client && hud_used)
+		hud_used.grue_hud()
+
+//health indicator
+		if (health >= maxHealth)
+			healths.icon_state = "health0"
+		else if (health >= 4*maxHealth/5)
+			healths.icon_state = "health1"
+		else if (health >= 3*maxHealth/5)
+			healths.icon_state = "health2"
+		else if (health >= 2*maxHealth/5)
+			healths.icon_state = "health3"
+		else if (health >= 1*maxHealth/5)
+			healths.icon_state = "health4"
+		else if (health > 0)
+			healths.icon_state = "health5"
+		else
+			healths.icon_state = "health6"
+//darkness level indicator
+		if (dark_dim_light==0)
+			healths2.icon_state= "lightlevel_dark"
+			healths2.name="nice and dark"
+		else if (dark_dim_light==1)
+			healths2.icon_state= "lightlevel_dim"
+			healths2.name="adequately dim"
+		else if (dark_dim_light==2)
+			healths2.icon_state= "lightlevel_bright"
+			healths2.name="painfully bright"
+
+/mob/living/simple_animal/hostile/grue/Life()
+	..()
+
+	//process shadow power and health according to current tile brightness level
+	if (stat!=DEAD)
+		if(isturf(loc))
+			var/turf/T = loc
+			current_brightness=10*T.get_lumcount()
+		else												//else, there's considered to be no light
+			current_brightness=0
+//		visible_message("<span class='warning'>\The [src] is in brightness level [current_brightness] with [health] health and [shadowpower] shadowpower.</span>") //debug
```

Comment:
No dead comments. Maybe make this proc every 300 life ticks (there is a var for that) if spam is an issue.
```dm
if ( (lifetick % 300) == 0)
    ...
```

Reply:
Sorry, that was just for debugging.